src/plugins/intel_npu/tests/unit/npuw/serialization.cpp

```diff
@@ -0,0 +1,279 @@
+}
+
+TEST(SerializationTest, OVTypes_Any) {
+    using namespace ov::npuw::s11n;
+
+    std::vector<ov::Any> var;
+    var.push_back(42);
+    var.push_back("42");
+    var.push_back(float(3.14));
+    var.push_back(true);
+    std::vector<ov::Any> res;
+    res.resize(var.size());
+
+    std::stringstream ss;
+
+    for (std::size_t i = 0; i < var.size(); ++i) {
+        write_any(ss, var[i]);
+        read_any(ss, res[i]);
+        EXPECT_EQ(var[i], res[i]);
+    }
+}
+
+TEST(SerializationTest, BasicTypes_array) {
+    using namespace ov::npuw::s11n;
+
+    IndicatorType res;
+
+    std::stringstream ss;
+
+    write(ss, NPUW_SERIALIZATION_INDICATOR);
+    read(ss, res);
+
+    EXPECT_EQ(NPUW_SERIALIZATION_INDICATOR, res);
+}
+
+TEST(SerializationTest, BasicTypes_pair) {
+    using namespace ov::npuw::s11n;
+
+    std::pair<int, float> var{42, 3.14};
+    std::pair<int, float> res;
+
+    std::stringstream ss;
+
+    write(ss, var);
+    read(ss, res);
+
+    EXPECT_EQ(var, res);
+}
+
+TEST(SerializationTest, BasicTypes_vector) {
+    using namespace ov::npuw::s11n;
+
+    std::vector<int> var{1, 2, 3, 45};
+    std::vector<int> res;
+
+    std::stringstream ss;
+
+    write(ss, var);
+    read(ss, res);
+
+    EXPECT_EQ(var, res);
+}
+
+TEST(SerializationTest, BasicTypes_map) {
+    using namespace ov::npuw::s11n;
+
+    std::map<int, std::string> var{{1, "a"}, {2, "b"}};
+    std::map<int, std::string> res;
+
+    std::stringstream ss;
+
+    write(ss, var);
+    read(ss, res);
+
+    EXPECT_EQ(var, res);
+}
+
+TEST(SerializationTest, BasicTypes_unordered_set) {
+    using namespace ov::npuw::s11n;
+
+    std::unordered_set<std::string> var{"a", "b", "c"};
+    std::unordered_set<std::string> res;
+
+    std::stringstream ss;
+
+    write(ss, var);
+    read(ss, res);
+
+    EXPECT_EQ(var, res);
+}
+
+TEST(SerializationTest, BasicTypes_optional) {
+    using namespace ov::npuw::s11n;
+
+    std::optional<int> var = 1;
+    std::optional<int> res;
+
+    std::optional<int> var2 = std::nullopt;
+    std::optional<int> res2;
+
+    std::stringstream ss;
+
+    write(ss, var);
+    write(ss, var2);
+    read(ss, res);
+    read(ss, res2);
+
+    EXPECT_EQ(var, res);
+    EXPECT_EQ(var2, res2);
+}
+
+TEST(SerializationTest, OVTypes_Tensor_with_weights) {
+    using namespace ov::npuw::s11n;
+
+    std::vector<uint8_t> data {0, 1, 2, 3};
+    ov::Tensor var(ov::element::u8, ov::Shape({2, 2}), data.data());
+    std::vector<ov::Tensor> res;
+
+    std::stringstream ss;
+
+    std::unordered_map<const void*, std::size_t> const_offset;
+    Context ctx(false, const_offset);
+
+    write_weightless(ss, {var}, ctx);
+    read_weightless(ss, res, nullptr);
+
+    EXPECT_EQ(res[0].get_element_type(), ov::element::u8);
+    EXPECT_EQ(res[0].get_shape(), ov::Shape({2, 2}));
+
+    std::vector<uint8_t> data_res(4, 0);
+    std::memcpy(data_res.data(), res[0].data(), 4);
+
+    EXPECT_EQ(data, data_res);
+}
```

Comment:
These all tests can be parameterized, might be it is needed to be done some day, so file will not grow endless.

Reply:
Added a fixme